starlette/testclient.py

```diff
@@ -77,6 +77,33 @@ def __init__(self, session: WebSocketTestSession) -> None:
         self.session = session
 
 
+class WebSocketDenial(Exception):
+    """Base exception for websocket denial responses."""
+
+
+class WebSocketDenialClose(WebSocketDisconnect, WebSocketDenial):
+    """A special case of WebSocketDisconnect, raised when connection
+    is rejected before being accepted with a websocket.close()."""
+
+
+class WebSocketDenialResponse(WebSocketDenial):
+    """
+    A special case of WebSocketDisconnect, raised in the TestClient if the
+    with a send_denial_response()
+    or a websocket.close()
+    """
+
+    def __init__(
+        self,
+        response_status: int,
+        response_headers: typing.List[typing.Tuple[bytes, bytes]] = [],
```

Comment:
The `response.headers` from the `TestClient` are a dictionary, right? Can we make this as well? (please confirm)

Reply:
the `httpx.Response` returned by TestClient contains headers of this type:
```
HeaderTypes = Union[
    "Headers",
    Mapping[str, str],
    Mapping[bytes, bytes],
    Sequence[Tuple[str, str]],
    Sequence[Tuple[bytes, bytes]],
]
```
The WebsocketDenialResponse contains the raw http response, thus a sequence of bytes tuples.
We could of course add a helper function to return a str:str mapping, with utf8 decoding, if you think that is worthwhile.

Hm, I guess we could even subclass httpx.Response for this case, rather than write our own.  I"ll try that.